rdf-delta-server-local/src/main/java/org/seaborne/delta/zk/direct/ConnectString.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  See the NOTICE file distributed with this work for additional
+ *  information regarding copyright ownership.
+ */
+
+package org.seaborne.delta.zk.direct;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ * Parses a config to create a connectString.
+ * </p>
+ *
+ * <p>
+ * The following ZooKeeper Ensemble config
+ * </p>
+ *
+ * <blockquote>
+ * server.1=localhost:2182:2183:participant;0.0.0.0:2181<br/>
+ * server.2=localhost:2282:2283:participant;0.0.0.0:2281<br/>
+ * server.3=localhost:2382:2383:participant;0.0.0.0:2381<br/>
+ * version=700000008
+ * </blockquote>
+ *
+ * <p>
+ * would be parsed into the following connectString:
+ * </p>
+ *
+ * <blockquote>
+ * localhost:2181,localhost:2281,localhost:2381
+ * </blockquote>
+ */
+public final class ConnectString implements CharSequence {
```

Comment:
Inhereting CharSequence feels like pushing the boundaries of Liskov substitutability, for what gain? Overly clever.